### PR TITLE
add Alias method for updating email address

### PR DESCRIPTION
**Pull request by WesleyDRobinson**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -30,6 +31,16 @@ Iterable.ensure(function(msg, settings){
 });
 
 /**
+ * We require both IDs be emails for alias to work in iterable
+ */
+
+Iterable.ensure(function(msg, settings){
+  if (msg.action() !== 'alias') return;
```
Comment (dontgitit) [inline lib/index.js]: Looks fine but hard to read... personally would prefer something like

```
ensure(...) {
  if (msg.action() === 'alias' && !(isEmail(...) && isEmail(..))) 
    return this.reject(...)
  return; 
}

**Comment by WesleyDRobinson** [comment]
Thank you!! @dontgitit  
I was going crazy on that test. No longer hard coded test 👍 

@hankim813 @sperand-io for final review

**Comment by hankim813** [comment]
@WesleyDRobinson @dontgitit 

> Not sure if worth mentioning, but at the moment, we don't allow changing your email if the new email already exists. Not sure how that would play with your typical use case.

Hmm if this is the case, what's the purpose of this PR? Isn't this `.alias()` call meant to update emails? Also we don't necessarily want users to send `email` as the userIds... that's against our spec.

**Comment by sperand-io** [comment]
Hey all — sorry for the delay here. 

I'm -1 on this addition (despite initially greenlighting @WesleyDRobinson on it) because the alias method is meant for changing a distinct userId, not for updating an email on an existing user. The supported way to update a users email in a tool that keys off userId is to just send a new email value. Wesley has indicated that iterable is moving towards supporting keying of userId, which is the correct long term solution to this issue.

In general, the alias method only exists because some tools don't do implicit aliasing from anonymous to known users, and because we don't (yet) have the ability to pave over that for them. Once we do, we'll likely phase out alias as a Segment method and do it transparently behind the scenes.

My biggest fear with this change is that invoking the alias method and forgetting to scope the call to iterable only could pollute or confuse other downstream tools, and at that point, since it's a strictly one-to-one segment to iterable API invocation, the end user might as well use iterable's API directly. It's a tricky case because we never want to suggest using a partner's API directly for something we can map, but when its a special case like this it's usually better to call directly into the tool :/

@WesleyDRobinson lmk what you think!

**Comment by dontgitit** [comment]
> Wesley has indicated that iterable is moving towards supporting keying of userId

FWIW, that's something we've been considering, but have no plans to implement yet.

**Pull request closed without merging**